### DISCUSSION TITLE
[Feature Request] Turning off softcore numbers, e.g., when displaying achievement unlock rates

### DISCUSSION THREAD

**Discussion by cIymax** [RetroAchievements/RAWeb#2782] [Ideas]
**A. Description**

1. In two separate website threads that generated quite a bit of discussion recently, RA users made the following suggestion.

Having a toggle to let users choose to hide softcore numbers and suppress softcore numbers from being included in any calculations involving hardcore numbers.  This applies to things like achievement unlock rates, achievement distributions, etc.

Under this proposal, if the toggle is in effect, then:
![image](https://github.com/user-attachments/assets/e5ddcf95-1882-40ef-a6cf-8f9299d993d7)
would become:
```
123 of 667
18.44% unlock rate
```
with no gray bar being shown.

2. Not mentioned was whether the toggle should default to ON, so I assume that even defaulting to OFF should be acceptable to the suggesting users.

3. If a toggle cannot be provided, it was further suggested that the numbers should at least be flipped.  Reason given was that the hardcore numbers are the ones of relative significance.
```
123 (159) of 667
18.44% (23.84) unlock rate
```
with the gray bar being shown.

4. Also not mentioned was whether there should be a user option to hide hardcore numbers.  I mention it here for sake of completeness.
```
159 of 667
23.84% unlock rate
```
with no yellow bar being shown.

5. I don't think the users asked for mixed-core calculations on the softcore side to exclude hardcore numbers, but an example of such exclusion would be the following.
```
36 of 667
5.40% unlock rate
```
with no yellow bar being shown.

6. The denominator could be adapted as well.
For instance, instead of "123 of 667" in the first proposed example, we could have the following.
```
123 of 631
19.49% unlock rate
```

**B. References**

Current thread: "Softcore should exist, but I should have the ability to not see it. :)"
https://retroachievements.org/viewtopic.php?t=28575

Original thread (locked by moderation): "Softcore shouldn't exist.  [But because it does, I should have the ability to not see it.]"
https://retroachievements.org/viewtopic.php?t=28500

**Comment by Jamiras** [comment]
I'm strongly against trying to eradicate/hide softcore. The driving argument in the forum posts is that having softcore unlocks shown obscures the difficulty of the achievement. 

As has been discussed many times in regards to retro ratio, unlock rate often reflects attrition more than difficulty. A very simple endgame achievement may have an arbitrarily low unlock rate just because people quit playing before they get to it.

If hiding the gray bar eliminates (roughly) half of the players for a set, the achievement automatically looks twice as difficult as it is "183 (63) of 3,235". would become "63 of 3,235 (1.9%)", when in reality it's probably closer to "63 of 1,832 (3.4%)". 

And if we try to shrink the bar to only account for the hardcore players, then we lose the players that loaded the game but never earned any achievements (notice how most games don't have any achievements with 100% unlock rate?).

-----

The data doesn't actually separate the users into hardcore/softcore buckets. Hardcore is just a tier above softcore. A user who has unlocked an achievement in hardcore has also implicitly unlocked it in softcore. For "183 (63) of 3,235", there's only actually 120 pure softcore unlocks. The bar is stacked:
![image](https://github.com/user-attachments/assets/d68f7173-66cd-43e5-98dc-11856ea5aabe)

Just like the achievement distribution graph:
![image](https://github.com/user-attachments/assets/f84235f0-d137-4774-b0cd-0612eec6dafe)

Which would (presumably) also be affected by hiding the softcore data.

-----

This sort of change would also affect things outside of the web page. In the last year, we've started sending unlock rates to clients to display when the achievement pops. It's using the same math as the webpage where hardcore unlocks have an intrinsically lower percentage because of the portion of users who have only unlocked the achievement in softcore. 

Rather than seeing the bar as "softcore players are ruining my stats", I see it as "this many players chose to have help". I still want to know how many people (of _all the people who played the game_) were able to do it without help, and how many were able to do it at all. I haven't tried it recently, but in the early 2000s, I still couldn't beat Battletoads, even with save states.
